src/agent/onefuzz-supervisor/src/coordinator.rs

```diff
@@ -54,21 +54,49 @@ pub struct NodeEventEnvelope {
 }
 
 #[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
-#[serde(rename_all = "snake_case", untagged)]
+#[serde(rename_all = "snake_case")]
 pub enum NodeEvent {
-    StateUpdate { state: NodeState },
-    WorkerEvent { event: WorkerEvent },
+    StateUpdate(StateUpdateEvent),
```

Comment:
'StateUpdateEvent' to be consistent with the other case WorkerEvent

Reply:
I'm explicitly leaving the variant name as `StateUpdate` so the external tag when serialized is `state_update`. I only include `Event` in `WorkerEvent` because it truly contains a nested, entirely different event type as its event data.

At one point I had `NodeEvent::WorkerEvent` as just `NodeEvent::Worker`, but it sort of had confusingly less context at the use site (because you'd get at it as something like `node_event.worker`).